test/unit/puppetlabs/services/request_handler/request_handler_core_test.clj

```diff
@@ -311,9 +311,13 @@
         (jruby-bootstrap/with-puppetserver-running-with-services
          app services {:jruby-puppet {:max-active-instances 1}}
          (jruby-testutils/wait-for-jrubies app)
-         (let [jruby-service (tk-app/get-service app :JRubyPuppetService)]
-           (future (testutils/get-catalog))
-           (is (deref first-promise 5000 false))
+         (let [in-catalog-request-future (promise)
+               jruby-service (tk-app/get-service app :JRubyPuppetService)]
+           (future
+            (deliver in-catalog-request-future true)
+            (testutils/get-catalog))
+           (deref in-catalog-request-future)
```

Comment:
I added in this bit just so that the timer doesn't include the overhead of launching/waiting for the future's thread to start.